Assets/Scripts/WorldProperties.cs

```diff
@@ -92,22 +90,100 @@ void Start()
         }
 
         /// <summary>
-        /// Recursively adds the clipShader to the parent and all its children
+        /// Cycle through the connected drones
         /// </summary>
-        /// <param name="parent">The topmost container of the objects which will have the shader added to them</param>
-        public static void AddClipShader(Transform parent)
+        public static void SelectNextDrone()
         {
-            if (parent.GetComponent<Renderer>())
+            Debug.Log("Selection next drone");
+
+            if (selectedDrone != null)
+            {
+                dronesQueue.Enqueue(selectedDrone);
+            }
+
+            if (dronesQueue.Count > 0)
             {
-                parent.GetComponent<Renderer>().material.shader = clipShader;
+                Drone nextDrone = dronesQueue.Dequeue();
```

Comment:
do you not enque the drone back onto the end?

Reply:
it's what line 99 does